File: src/main/java/com/lessmarkup/interfaces/security/CurrentUser.java
package com.lessmarkup.interfaces.security;

import com.google.gson.JsonObject;
import com.lessmarkup.interfaces.data.DomainModel;
import com.lessmarkup.interfaces.structure.Tuple;
import java.security.NoSuchAlgorithmException;
import java.sql.SQLException;
import java.util.List;
import java.util.OptionalLong;

public interface CurrentUser {
    OptionalLong getUserId();
    List<Long> getGroups();
    JsonObject getProperties();
    boolean isAdministrator();
    boolean isApproved();
    boolean isFakeUser();
    boolean emailConfirmed();
    String getEmail();
    String getUserName();
    void logout();
    void refresh();
    boolean loginWithPassword(String email, String password, boolean savePassword, boolean allowAdmin, boolean allowRegular, String encodedPassword);
    boolean loginWithOAuth(String provider, String providerUserId, boolean savePassword, boolean allowAdmin, boolean allowRegular);
    void deleteSelf(String password) throws Exception;
    boolean checkPassword(DomainModel domainModel, String password) throws SQLException;
    Tuple<String, String> getLoginHash(String email) throws NoSuchAlgorithmException, SQLException;
}

File: src/main/scala/com/lessmarkup/interfaces/security/CurrentUser.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.security

import com.google.gson.JsonObject
import com.lessmarkup.interfaces.data.DomainModel

trait CurrentUser {
  def getUserId: Option[Long]

  def getGroups: Option[Seq[Long]]

  def getProperties: Option[JsonObject]

  def isAdministrator: Boolean

  def isApproved: Boolean

  def isFakeUser: Boolean

  def emailConfirmed: Boolean

  def getEmail: Option[String]

  def getUserName: Option[String]

  def logout()

  def refresh()

  def loginWithPassword(email: String, password: String, savePassword: Boolean, allowAdmin: Boolean, allowRegular: Boolean, encodedPassword: String): Boolean

  def loginWithOAuth(provider: String, providerUserId: String, savePassword: Boolean, allowAdmin: Boolean, allowRegular: Boolean): Boolean

  def deleteSelf(password: String)

  def checkPassword(domainModel: DomainModel, password: String): Boolean

  def getLoginHash(email: String): (String, String)
}